lib/css_parser.rb

```diff
@@ -145,23 +134,24 @@ def self.calculate_specificity(selector)
   #               "http://example.org/style/basic.css").inspect
   #  => "body { background: url('http://example.org/style/yellow.png?abc=123') };"
   def self.convert_uris(css, base_uri)
-    base_uri = Addressable::URI.parse(base_uri) unless base_uri.kind_of?(Addressable::URI)
+    base_uri = Addressable::URI.parse(base_uri) unless base_uri.is_a?(Addressable::URI)
 
     css.gsub(URI_RX) do
-      uri = $1.to_s
-      uri.gsub!(/["']+/, '')
+      uri = Addressable::URI.parse(Regexp.last_match(1).to_s.gsub(/["']+/, ''))
       # Don't process URLs that are already absolute
-      unless uri =~ /^[a-z]+\:\/\//i
+      if uri.relative?
```

Comment:
prefer to leave this alone, looks different ...

Reply:
Yep, but right now we're passing `data:`, for example, and it works just because addressable convert `http_uri + data_uri` to `data_uri`. I think that `.relative?` check is the proper thing to do, but it could be a separate commit.